data-prepper-plugins/kafka-source/README.md

```diff
@@ -0,0 +1,150 @@
+              key_deserializer: org.apache.kafka.common.serialization.StringDeserializer
+              value_deserializer: org.apache.kafka.common.serialization.StringDeserializer
+              schema_type: plaintext
+              record_type: plaintext
+            ssl:
+              ssl_truststore_location: C:/Users/AM20395480/kafka.jksa
+              ssl_truststore_password: kafkasource
+              ssl_keystore_location: C:/Users/AM20395480/Music/certificates/identity.jks
+              ssl_keystore_password: kafkasource
+              ssl_key_password: kafkasource  
+         - topic:
+            name: my-topic-1
+            auth_type: sasl_ssl
+            consumer_group:
+              group_name: kafka-consumer-group-1
+              group_id: DPKafkaProj-1
+              workers: 10
+            schema:
+              registry_url: http://localhost:8081/
+              key_deserializer: org.apache.kafka.common.serialization.StringDeserializer
+              value_deserializer: org.apache.kafka.common.serialization.StringDeserializer
+              schema_type: plaintext
+              record_type: plaintext
+            ssl:
+              ssl_truststore_location: C:/Users/AM20395480/kafka.jksa
+              ssl_truststore_password: kafkasource
+              ssl_keystore_location: C:/Users/AM20395480/Music/certificates/identity.jks
+              ssl_keystore_password: kafkasource
+              ssl_key_password: kafkasource  
+  sink:
+    - stdout:
+```
+
+## Configuration
+
+- `bootstrap_servers` (Required) : It is a host/port to use for establishing the initial connection to the Kafka cluster.
+
+- `topics` (Required) : The topic in which kafka source plugin associated with to read the messages.The maximum number of topics should be 10.
+
+- `topic name` (Required) : This denotes the name of the topic and it is a mandatory one.
+
+- `auth_type` (Optional) : The auth_type directive selects the method that is used to authenticate the user. There is no auth type by default. The most common supported methods are SASL_PLAINTEXT,SASL_SSL,SASL_OAUTH
+
+- `group_name`  (Required) : A consumer group name which this kafka consumer belongs to. 
+
+- `group_id` (Required) : A consumer group id which this kafka consumer is associated with. 
+
+- `workers` (Optional) : Number of multithreaded consumers associated with each topic. Defaults to `10` and its maximum value should be 200.
+
+- `autocommit` (Optional) : If false the consumer's offset will not be periodically committed in the background. Defaults to `false`.
+
+- `autocommit_interval` (Optional) : The frequency in seconds that the consumer offsets are auto-committed to Kafka. Defaults to `1s`.
+
+- `session_timeout` (Optional) : The timeout used to detect client failures when using Kafka's group management. It is used for the rebalance.
+
+- `max_retry_attempts` (Optional) : The maximum attempts to retry a failed write operation to the BUFFER. Defaults to `INFINITE`.
```

Comment:
We should make it explicit that BUFFER here is data prepper buffer. Also, this config is independent of kafka topic.